[vcpkg_fixup_pkgconfig] build failure with "Could not find PKGCONFIG using the following names: pkg-config"

**Host Environment**
- OS: Microsoft Windows [Version 10.0.19041.388]
- Compiler: 
   - Visual Studio 2017 Version: 15.9.25  VS Tools Version: 14.16.27023 
   - Visual Studio 2019 Version: 16.6.5  VS Tools Version: 14.26.28801 

**To Reproduce**
Steps to reproduce the behavior:
After latest update to vcpkg #11550 Commit https://github.com/microsoft/vcpkg/commit/458c20e94b9f558b523f187ce403bd062e131cdb

set VCPKG_DEFAULT_TRIPLET=x64-windows

1. `./vcpkg upgrade --no-dry-run`

2. `./vcpkg install bzip2`

**Failure logs** 
- Cut and past the appropriate build messages from the console output.
1. vcpkg upgrade console log
`The following packages will be rebuilt:`
`  * bzip2[core]:x64-windows`
`Additional packages (*) will be modified to complete this operation.`
`Starting package 1/2: bzip2:x64-windows`
`Removing package bzip2:x64-windows...`
`Removing package bzip2:x64-windows... done`
`Elapsed time for package bzip2:x64-windows: 6.266 ms`
`-- Using cached D:/Git/vcpkg.downloads/bzip2-1.0.6.tar.gz`
`-- Cleaning sources at D:/Git/vs2019.test/buildtrees/bzip2/src/1.0.6-8838f21f8e.clean. Use --editable to skip cleaning for the packages you specify.`
`-- Extracting source D:/Git/vcpkg.downloads/bzip2-1.0.6.tar.gz`
`-- Applying patch fix-import-export-macros.patch`
`-- Applying patch fix-windows-include.patch`
`-- Using source at D:/Git/vs2019.test/buildtrees/bzip2/src/1.0.6-8838f21f8e.clean`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`-- Building x64-windows-rel`
`-- Fixing pkgconfig`
`CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:279 (find_program):`
**`  Could not find PKGCONFIG using the following names: pkg-config`**
`Call Stack (most recent call first):`
`  ports/bzip2/portfile.cmake:51 (vcpkg_fixup_pkgconfig)`
`  scripts/ports.cmake:79 (include)`

`Error: Building package bzip2:x64-windows failed with: BUILD_FAILED`
`Elapsed time for package bzip2:x64-windows: 2.98 s`

`Total elapsed time: 4.544 s`

`RESULTS`
`    bzip2:x64-windows: vcpkg::Commands::Build::BuildResult_NULLVALUE: 6.266 ms`
`    bzip2:x64-windows: BUILD_FAILED: 2.98 s`

`SUMMARY`
`    vcpkg::Commands::Build::BuildResult_NULLVALUE: 1`
`    SUCCEEDED: 0`
`    BUILD_FAILED: 1`
`    POST_BUILD_CHECKS_FAILED: 0`
`    FILE_CONFLICTS: 0`
`    CASCADED_DUE_TO_MISSING_DEPENDENCIES: 0`
`    EXCLUDED: 0`

2. vcpkg instal bzip2:x64-windows
`Computing installation plan...`
`The following packages will be built and installed:`
`    bzip2[core]:x64-windows`
`Detecting compiler hash for triplet x64-windows...`
`Starting package 1/1: bzip2:x64-windows`
`Building package bzip2[core]:x64-windows...`
`-- Using cached D:/Git/vcpkg.downloads/bzip2-1.0.6.tar.gz`
`-- Extracting source D:/Git/vcpkg.downloads/bzip2-1.0.6.tar.gz`
`-- Applying patch fix-import-export-macros.patch`
`-- Applying patch fix-windows-include.patch`
`-- Using source at D:/Git/vs2017.test/buildtrees/bzip2/src/1.0.6-8838f21f8e.clean`
`-- Configuring x64-windows`
`-- Building x64-windows-dbg`
`-- Building x64-windows-rel`
`-- Fixing pkgconfig`
`CMake Error at scripts/cmake/vcpkg_fixup_pkgconfig.cmake:279 (find_program):`
**`  Could not find PKGCONFIG using the following names: pkg-config`**
`Call Stack (most recent call first):`
`  ports/bzip2/portfile.cmake:51 (vcpkg_fixup_pkgconfig)`
`  scripts/ports.cmake:79 (include)`

`Error: Building package bzip2:x64-windows failed with: BUILD_FAILED`
`Please ensure you're using the latest portfiles with `.\vcpkg update`, then`
`submit an issue at https://github.com/Microsoft/vcpkg/issues including:`
`  Package: bzip2:x64-windows`
`  Vcpkg version: 2020.06.15-nohash`

`Additionally, attach any relevant sections from the log files above.`

- Please attach any additional failure logs mentioned in the console output.
No failure in logs, appears to compile/link properly, just fails to install due to error listed.

**Additional context**

I tried applying both fixes. The first fix regarding `vcpkg_fixup_pkconfig.cmake` worked for me, but apparently the second one didn't.
Error description:
```
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:72 (message):
    Command failed: C:/dev/vcpkg/downloads/tools/msys2/msys64/usr/bin/bash.exe --noprofile --norc -c "pacman -S --noconfirm --needed pkg-config"
    Working Directory: C:/dev/vcpkg/downloads/tools/msys2
    Error code: 1
    See logs for more information:
      C:\dev\vcpkg\buildtrees\bzip2\msys-pacman-x86-windows-err.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_acquire_msys.cmake:149 (vcpkg_execute_required_process)
  scripts/cmake/vcpkg_fixup_pkgconfig.cmake:280 (vcpkg_acquire_msys)
  ports/bzip2/portfile.cmake:51 (vcpkg_fixup_pkgconfig)
  scripts/ports.cmake:79 (include)


Error: Building package bzip2:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: bzip2:x86-windows
  Vcpkg version: 2020.06.15-nohash
```
And the other error log, made by pacman (`msys-pacman-x86-windows-err.log`):
```
error: mingw32: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: mingw64: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: msys: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" is unknown
error: key "4A6129F4E4B84AE46ED7F635628F528CF3053E04" could not be looked up remotely
error: database 'mingw32' is not valid (invalid or corrupted database (PGP signature))
error: database 'mingw64' is not valid (invalid or corrupted database (PGP signature))
error: database 'msys' is not valid (invalid or corrupted database (PGP signature))
```

--- a/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
+++ b/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
@@ -276,7 +276,7 @@ function(vcpkg_fixup_pkgconfig)
     endif()
 
     if(NOT PKGCONFIG)
-        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/Cellar/pkg-config/0.29.2_3" REQUIRED)
+        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/Cellar/pkg-config/0.29.2_3")
         if(NOT PKGCONFIG AND CMAKE_HOST_WIN32)
             vcpkg_acquire_msys(MSYS_ROOT PACKAGES pkg-config)
             find_program(PKGCONFIG pkg-config PATHS "${MSYS_ROOT}/usr/bin" REQUIRED)

--- a/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
+++ b/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
@@ -276,7 +276,7 @@ function(vcpkg_fixup_pkgconfig)
     endif()
 
     if(NOT PKGCONFIG)
-        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/bin" "/usr/local/Cellar/pkg-config/0.29.2_3")
+        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/bin")
         if(NOT PKGCONFIG AND CMAKE_HOST_WIN32)
             vcpkg_acquire_msys(MSYS_ROOT PACKAGES pkg-config)
             find_program(PKGCONFIG pkg-config PATHS "${MSYS_ROOT}/usr/bin" REQUIRED)

--- a/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
+++ b/scripts/cmake/vcpkg_fixup_pkgconfig.cmake
@@ -276,7 +276,7 @@ function(vcpkg_fixup_pkgconfig)
     endif()
 
     if(NOT PKGCONFIG)
-        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/Cellar/pkg-config/0.29.2_3")
+        find_program(PKGCONFIG pkg-config PATHS "bin" "/usr/bin" "/usr/local/bin" "/usr/local/Cellar/pkg-config/0.29.2_3")
         if(NOT PKGCONFIG AND CMAKE_HOST_WIN32)
             vcpkg_acquire_msys(MSYS_ROOT PACKAGES pkg-config)
             find_program(PKGCONFIG pkg-config PATHS "${MSYS_ROOT}/usr/bin" REQUIRED)

